// eslint-config-airbnb-extended/dist/legacy/rules/typescript/typescriptSettings.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
const utils_1 = require("../../../utils");
const legacyTypescriptSettingsRules = {
    name: 'airbnb/config/typescript/settings/legacy',
    files: utils_1.allFiles,
    settings: {
        // Apply special parsing for TypeScript files
        'import/parsers': {
            '@typescript-eslint/parser': utils_1.tsExtensionsResolver,
        },
        // Append 'ts' extensions to Airbnb 'import/resolver' setting
        // Original: ['.mjs', '.js', '.json']
        'import/resolver': {
            node: {
                extensions: [...utils_1.tsExtensionsResolver, '.json'],
            },
            typescript: true,
        },
        // Append 'ts' extensions to Airbnb 'import/extensions' setting
        // Original: ['.js', '.mjs', '.jsx']
        'import/extensions': utils_1.tsExtensionsResolver,
        // Resolve type definition packages
        'import/external-module-folders': ['node_modules', 'node_modules/@types'],
    },
};
exports.default = legacyTypescriptSettingsRules;
