// eslint-config-airbnb-extended/dist/extensions/base/recommended.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const getImportSettings_1 = __importDefault(require("../../helpers/getImportSettings"));
const getStylisticLegacyConfig_1 = __importDefault(require("../../helpers/getStylisticLegacyConfig"));
const utils_1 = require("../../utils");
const baseRecommendedExtensionsConfig = [
    {
        name: 'airbnb/config/base-configurations',
        files: utils_1.allFiles,
        languageOptions: {
            parserOptions: {
                ecmaVersion: 2018,
                sourceType: 'module',
            },
        },
        settings: {
            'import-x/core-modules': [],
            'import-x/ignore': ['node_modules', String.raw `\.(coffee|scss|css|less|hbs|svg|json)$`],
        },
    },
    {
        name: 'airbnb/config/base-settings-extensions-configurations',
        files: utils_1.jsFileWithoutReact,
        settings: (0, getImportSettings_1.default)({ javascript: true, typescript: false, jsx: false }),
    },
    Object.assign({ name: 'airbnb/config/base-disable-legacy-stylistic-js-config', files: utils_1.allFiles }, (0, getStylisticLegacyConfig_1.default)('javascript')),
];
exports.default = baseRecommendedExtensionsConfig;
