// eslint-config-airbnb-extended/dist/plugins/index.js
"use strict";
/* eslint-disable @typescript-eslint/no-require-imports, @typescript-eslint/no-unsafe-return, unicorn/prefer-module */
Object.defineProperty(exports, "__esModule", { value: true });
/**
 * as is given due to less size of index.d.ts
 */
const plugins = {
    get stylistic() {
        return require('../plugins/stylisticPlugin').default;
    },
    get importX() {
        return require('../plugins/importXPlugin').default;
    },
    get node() {
        return require('../plugins/nodePlugin').default;
    },
    get react() {
        return require('../plugins/reactPlugin').default;
    },
    get reactA11y() {
        return require('../plugins/reactA11yPlugin').default;
    },
    get reactHooks() {
        return require('../plugins/reactHooksPlugin').default;
    },
    get next() {
        return require('../plugins/nextPlugin').default;
    },
    get typescriptEslint() {
        return require('../plugins/typescriptEslintPlugin').default;
    },
};
exports.default = plugins;
