// eslint-config-airbnb-extended/dist/legacy/rules/react/reactHooks.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const eslint_plugin_react_hooks_1 = __importDefault(require("eslint-plugin-react-hooks"));
const utils_1 = require("../../../utils");
const legacyReactHooksRules = {
    name: 'airbnb/config/react-hooks/legacy',
    files: utils_1.allFiles,
    plugins: {
        'react-hooks': eslint_plugin_react_hooks_1.default,
    },
    languageOptions: {
        parserOptions: {
            ecmaFeatures: {
                jsx: true,
            },
        },
    },
    rules: {
        // Enforce Rules of Hooks
        // https://github.com/facebook/react/blob/c11015ff4f610ac2924d1fc6d569a17657a404fd/packages/eslint-plugin-react-hooks/src/RulesOfHooks.js
        'react-hooks/rules-of-hooks': 'error',
        // Verify the list of the dependencies for Hooks like useEffect and similar
        // https://github.com/facebook/react/blob/1204c789776cb01fbaf3e9f032e7e2ba85a44137/packages/eslint-plugin-react-hooks/src/ExhaustiveDeps.js
        'react-hooks/exhaustive-deps': 'error',
    },
};
exports.default = legacyReactHooksRules;
