// eslint-config-airbnb-extended/dist/utils/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.allFiles = exports.tsFileWithoutReact = exports.tsFiles = exports.jsFileWithoutReact = exports.jsFiles = exports.tsExtensionsRule = exports.tsExtensionsResolver = exports.tsExtensionsWithReactDTS = exports.tsExtensionsWithReact = exports.tsExtensions = exports.jsExtensionsRule = exports.jsExtensionsResolver = exports.jsExtensionsWithReact = exports.jsExtensions = void 0;
exports.jsExtensions = ['.js', '.cjs', '.mjs'];
exports.jsExtensionsWithReact = [...exports.jsExtensions, '.jsx'];
exports.jsExtensionsResolver = [...exports.jsExtensions, '.json'];
exports.jsExtensionsRule = Object.fromEntries(Object.values(exports.jsExtensionsWithReact).map((val) => [val.slice(1), 'never']));
exports.tsExtensions = ['.ts', '.cts', '.mts'];
exports.tsExtensionsWithReact = [...exports.tsExtensions, '.tsx'];
exports.tsExtensionsWithReactDTS = [...exports.tsExtensionsWithReact, '.d.ts'];
exports.tsExtensionsResolver = [...exports.tsExtensions, '.d.ts'];
exports.tsExtensionsRule = Object.fromEntries(Object.values(exports.tsExtensionsWithReact).map((val) => [val.slice(1), 'never']));
// ESLINT CONFIG FILES
exports.jsFiles = exports.jsExtensionsWithReact.map((val) => `**/*${val}`);
exports.jsFileWithoutReact = exports.jsExtensions.map((val) => `**/*${val}`);
exports.tsFiles = exports.tsExtensionsWithReactDTS.map((val) => `**/*${val}`);
exports.tsFileWithoutReact = exports.tsExtensionsResolver.map((val) => `**/*${val}`);
exports.allFiles = [...exports.jsFiles, ...exports.tsFiles];
