// eslint-config-airbnb-extended/dist/legacy/configs/base/legacy.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const globals_1 = __importDefault(require("globals"));
const config_1 = __importDefault(require("../../../legacy/configs/base/config"));
const utils_1 = require("../../../utils");
const legacyBaseLegacyConfig = [
    config_1.default.bestPractices,
    config_1.default.errors,
    config_1.default.node,
    config_1.default.style,
    config_1.default.variables,
    {
        name: 'airbnb/config/base-legacy-configurations/legacy',
        files: utils_1.allFiles,
        languageOptions: {
            globals: Object.assign(Object.assign({}, globals_1.default.browser), globals_1.default.node),
        },
        rules: {
            'comma-dangle': ['error', 'never'],
            'prefer-numeric-literals': 'off',
            'no-restricted-properties': [
                'error',
                {
                    object: 'arguments',
                    property: 'callee',
                    message: 'arguments.callee is deprecated',
                },
                {
                    property: '__defineGetter__',
                    message: 'Please use Object.defineProperty instead.',
                },
                {
                    property: '__defineSetter__',
                    message: 'Please use Object.defineProperty instead.',
                },
            ],
            'no-var': 'off',
            'prefer-object-spread': 'off',
            strict: ['error', 'safe'],
        },
    },
];
exports.default = legacyBaseLegacyConfig;
